Mathlib/Algebra/BigOperators/Fin.lean

```diff
@@ -181,6 +171,73 @@ theorem prod_trunc {M : Type*} [CommMonoid M] {a b : ℕ} (f : Fin (a + b) → M
   rw [prod_univ_add, Fintype.prod_eq_one _ hf, mul_one]
   rfl
 
+section Interval
+
+variable {M : Type*} [CommMonoid M] {n : ℕ} {v : Fin (n + 1) → M}
+
+@[to_additive]
+theorem prod_Ioi_zero : ∏ i ∈ Ioi 0, v i = ∏ j : Fin n, v j.succ := by
+  rw [Ioi_zero_eq_map, Finset.prod_map, val_succEmb]
+
+@[to_additive]
+theorem prod_Ioi_succ (i : Fin n) : ∏ j ∈ Ioi i.succ, v j = ∏ j ∈ Ioi i, v j.succ := by
+  rw [Ioi_succ, Finset.prod_map, val_succEmb]
+
+@[to_additive (attr := simp)]
+theorem prod_Ici_zero : ∏ i ∈ Ici 0, v i = ∏ j : Fin (n + 1), v j := by
+  rw [Ici_eq_Icc, ← bot_eq_zero, Icc_bot_top]
+
+@[to_additive]
+theorem prod_Ici_succ (i : Fin n) :
+    ∏ j ∈ Ici i.succ, v j = ∏ j ∈ Ici i, v j.succ := by
+  rw [Ici_succ, Finset.prod_map, val_succEmb]
+
+@[to_additive (attr := simp)]
+theorem prod_Iio_zero : ∏ i ∈ Iio 0, v i = 1 := rfl
+
+@[to_additive]
+theorem prod_Iio_succ (i : Fin n) :
+    ∏ j ∈ Iio i.succ, v j = (∏ j ∈ Iio i.castSucc, v j) * v i.castSucc := by
+  calc
+    _ = ∏ j ∈ Icc 0 i.castSucc, v j := prod_congr rfl (fun _ _ => rfl)
+    _ = ∏ j ∈ insert i.castSucc (Ico 0 i.castSucc), v j :=
+      prod_congr (by simp only [zero_le, Ico_insert_right]) (fun _ _ => rfl)
+    _ = v i.castSucc * ∏ j ∈ Iio i.castSucc, v j :=
+      prod_insert (by simp only [mem_Ico, zero_le, lt_self_iff_false, and_false, not_false_eq_true])
+    _ = (∏ j ∈ Iio i.castSucc, v j) * v i.castSucc := mul_comm _ _
+
+@[to_additive]
+theorem prod_Iio_eq_univ (i : Fin (n + 1)) :
+    ∏ j ∈ Iio i, v j = ∏ j : Fin i, v (Fin.castLE i.isLt.le j) := by
+  induction i using Fin.induction with
+  | zero => simp only [prod_Iio_zero, val_zero, univ_eq_empty, prod_empty]
+  | succ i hi => simp only [prod_Iio_succ, hi, prod_univ_castSucc, val_succ]; congr
+
+@[to_additive (attr := simp)]
+theorem prod_Iic_zero : ∏ j ∈ Iic 0, v j = v 0 := by
```

Comment:
Iic 0 = {0} might also be enough, or you could keep this one as well for symmetry

Reply:
Yes, please prove `Iic 0 = {0}`